src/sentry/releasehealth/metrics.py

```diff
@@ -347,6 +358,137 @@ def _count_users(total: bool, referrer: str) -> Dict[Any, int]:
+            # init counter. This should take care of most cases on its own.
+            init_sessions_query = Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity(EntityKey.MetricsCounters.value),
+                select=select,
+                where=where
+                + [
+                    Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+                    Condition(
+                        Column(tag_key(org_id, "session.status")), Op.EQ, tag_value(org_id, "init")
+                    ),
+                ],
+            )
+
+            rows = raw_snql_query(
+                init_sessions_query,
+                referrer="releasehealth.metrics.get_release_sessions_time_bounds.init_sessions",
+                use_cache=False,
+            )["data"]
+        except MetricIndexNotFound:
+            rows = []
+
+        try:
+            # Take care of potential timestamp updates by looking at the metric
+            # for session duration, which is emitted once a session is closed ("terminal state")
+            #
+            # There is a testcase checked in that tests specifically for a
+            # session update that lowers session.started. We don't know if that
+            # testcase matters particularly.
+            terminal_sessions_query = Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity(EntityKey.MetricsDistributions.value),
+                select=select,
+                where=where
+                + [
+                    Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session.duration")),
+                ],
+            )
+            rows.extend(
+                raw_snql_query(
+                    terminal_sessions_query,
+                    referrer="releasehealth.metrics.get_release_sessions_time_bounds.terminal_sessions",
+                    use_cache=False,
+                )["data"]
+            )
+        except MetricIndexNotFound:
+            pass
+
+        # This check is added because if there are no sessions found, then the
+        # aggregations query return both the sessions_lower_bound and the
+        # sessions_upper_bound as `0` timestamp and we do not want that behaviour
+        # by default
+        # P.S. To avoid confusion the `0` timestamp which is '1970-01-01 00:00:00'
+        # is rendered as '0000-00-00 00:00:00' in clickhouse shell
+        formatted_unix_start_time = datetime.utcfromtimestamp(0).strftime("%Y-%m-%dT%H:%M:%S+00:00")
+
+        lower_bound = None
+        upper_bound = None
+
+        for row in rows:
+            if set(row.values()) == {formatted_unix_start_time}:
+                continue
+            if lower_bound is None or row["min"] < lower_bound:
+                lower_bound = row["min"]
+            if upper_bound is None or row["max"] > upper_bound:
+                upper_bound = row["max"]
```

Comment:
I am a bit confused by this.
I am not sure what the function really returns since we appear to mix the session start time and the session end time, so (If I get it ) we return something like: min and max times when a session for a particular release was seen ? 
That will most likely be something like  min(session_start), max(session_end) except for the sessions that are still running and we don't have a session end and for those we would have also a max(session_start).

If that's what the intention is than everything is fine.

Reply:
we are supposed to do `min(session_start), max(session_start)`. where do you see us using session_end?

Reply:
My impression is that the counter `session` with the "init" tag is pushed at the beginning of the session and the distribution `session.duration` is pushed at the end of the session so the timestamps for the two queries would represent a combination of the session starts and session ends... I might be missing something.